frontend/src/routes/dashboard/admin/index.tsx

```diff
@@ -1,12 +1,110 @@
+import type {
+  PaginatedRows,
+  PaginationParams,
+} from '~/components/table/server-paginated-data-table';
+import { ServerPaginatedDataTable } from '~/components/table/server-paginated-data-table';
+import { hybridPaginationHook, PaginationFetcher } from '~/hooks/hybridPaginationHook';
+
+export const serializeQueryUserPaginationParams = (paginationParams: PaginationParams) => {
+  const paramsForQuery: { [key: string]: string } = {
+    limit: '' + paginationParams.limit,
+    page: '' + paginationParams.page,
+  };
+  if (paginationParams.filter) {
+    paramsForQuery.filter = paginationParams.filter;
+  }
+  if (paginationParams.sortColumn && paginationParams.sort) {
+    paramsForQuery[`sort[${paginationParams.sortColumn}]`] = paginationParams.sort;
+  }
+
+  return new URLSearchParams(paramsForQuery).toString().replace(/%5B/g, '[').replace(/%5D/g, ']');
+};
 
 export default component$(() => {
+  const firstLoading = useStore({ value: true });
+
+  const fetchUserData: PropFunction<PaginationFetcher> = $(
+    async (paginationParams: PaginationParams) => {
+      const headers = {
+        method: 'GET',
+        headers: { 'Content-Type': 'application/json' },
+      };
+
+      const queryParams = serializeQueryUserPaginationParams(paginationParams);
+
+      const data = await authorizedFetch(
+        `${process.env.API_DOMAIN}/api/v1/users?${queryParams}`,
+        headers
+      );
+      return await data.json();
+    }
+  );
+
+  const { fetchRowsHandler } = hybridPaginationHook(fetchUserData);
+
+  useVisibleTask$(
+    () => {
+      firstLoading.value = false;
+    },
+    { strategy: 'document-ready' }
+  );
+
+  const usersResource = useResource$<{ data: PaginatedRows<UserCtx>; total: number }>(
+    async ({ track, cleanup }) => {
+      track(() => firstLoading.value);
+      if (firstLoading.value) return;
+      const abortController = new AbortController();
+      cleanup(() => abortController.abort('cleanup'));
+
+      const data = await authorizedFetch(
+        `${process.env.API_DOMAIN}/api/v1/users?limit=10&sort[name]=asc`,
+        {
+          signal: abortController.signal,
+          method: 'GET',
+          headers: { 'Content-Type': 'application/json' },
+        }
+      );
+      return await data.json();
+    }
+  );
+
   return (
-    <>
+    <div class="p-10">
       <h1>Admin panel</h1>
-      <p class="text-red-700">//TODO: Implement a table of users</p>
-    </>
+
+      <Resource
+        value={usersResource}
+        onPending={() => <p>Loading...</p>}
+        onRejected={() => <p>Failed to fetch users data</p>}
+        onResolved={(payload) => {
+          if (!Array.isArray(payload?.data)) return <p>Failed to fetch users data</p>;
+          const { data, total } = payload;
+          return (
+            <ServerPaginatedDataTable
+              rows={data}
+              totalRowCount={total}
+              emitFetchRows={fetchRowsHandler as any} //TODO check type issue
```

Comment:
What about this todo?

Reply:
this has to do with qwik types.
thb not sure how to properly convert or use the serialization types for functions passed to scopes

Reply:
Alright :)